src/controllers/trust.involved.controller.ts

```diff
@@ -0,0 +1,80 @@
+import { NextFunction, Request, Response } from 'express';
+import { TrusteeTypeChoice } from '../model/trustee.type.model';
+import * as config from '../config';
+import { logger } from '../utils/logger';
+
+const TRUST_INVOLVED_TEXTS = {
+  title: 'Individuals or entities involved in the trust',
+};
+
+const get = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): void => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    const templateName = config.TRUST_INVOLVED_PAGE;
+
+    return res.render(
+      templateName,
+      {
+        backLinkUrl: `${config.TRUST_DETAILS_URL}/${req.params['id']}`,
+        templateName,
+        pageParams: {
+          title: TRUST_INVOLVED_TEXTS.title,
+          checkYourAnswersUrl: config.CHECK_YOUR_ANSWERS_URL,
+        },
+      },
+    );
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+const post = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    if (req.body.noMoreToAdd) {
+      return res.redirect(config.CHECK_YOUR_ANSWERS_URL);
+    }
+
+    const typeOfTrustee = req.body.typeOfTrustee;
+
+    switch (typeOfTrustee) {
+        case TrusteeTypeChoice.historical:
+          logger.info("TODO: Route to trust-historical-beneficial-owner page ");
+          break;
+        case TrusteeTypeChoice.individual:
+          logger.info("TODO: Route to trust-individual page ");
+          break;
+        case TrusteeTypeChoice.legalEntity:
+          logger.info("TODO: Route to trust-ole page ");
+          break;
+        default:
+          logger.info("Error message and stay in same page ");
+    }
+    logger.info("For Current story, just re-displays the page (delete this comment and below line once all Trustee pages are done");
+    // the req.params['id'] is already validated in the has.trust.middleware
```

Comment:
Have you explored some other way of not just adding a comment but stopping the Sonar warning here? What if somebody removes the validation logic in `has.trust.middleware` - either by accident or deliberately - we'll continue ignoring the Sonar warning when we shouldn't. A code solution to this would be ideal.

Reply:
@dgolubevs @mouhajer-ch @markpit - what are your thoughts on this?

Reply:
I would try the sonar recommended fix,  check the url is not malicious before redirecting, by checking it starts with the path you are expecting

Reply:
@markpit - please review the latest (no sonar security error)

Reply:
You will need to handle what happens if the url is not secure/valid other wise it looks like the controller won't return anything and just hang. I would suggest displaying the error page by creating an error and calling next(error) (also make sure you log at error level so if this happens in live we can look in Kibana logs for an explanation)

Reply:
@markpit - I have changed the function to throw an error if invalid (which will never happen - remember we are doing this to keep Sonar sweet)

Reply:
looks ok to me now

Reply:
Looks good to me too -resolving comment. Thanks for fixing @SteveHicksCH .